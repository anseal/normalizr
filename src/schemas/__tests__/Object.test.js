// eslint-env jest
import { normalize, schema } from '../../'

describe(`${schema.Object.name} normalization`, () => {
	test('normalizes an object', () => {
		const userSchema = new schema.Entity('user')
		const object = new schema.Object({
			user: userSchema,
		})
		expect(normalize({ user: { id: 1 } }, object)).toMatchSnapshot()
	})

	test(`normalizes plain objects as shorthand for ${schema.Object.name}`, () => {
		const userSchema = new schema.Entity('user')
		expect(normalize({ user: { id: 1 } }, { user: userSchema })).toMatchSnapshot()
	})

	test('filters out undefined and null values', () => {
		const userSchema = new schema.Entity('user')
		const users = { foo: userSchema, bar: userSchema, baz: userSchema }
		expect(normalize({ foo: {}, bar: { id: '1' } }, users)).toMatchSnapshot()
	})
})
