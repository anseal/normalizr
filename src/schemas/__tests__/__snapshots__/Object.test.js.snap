// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObjectSchema normalization filters out undefined and null values 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": "1",
      },
      "undefined": Object {},
    },
  },
  "result": Object {
    "bar": "1",
  },
}
`;

exports[`ObjectSchema normalization normalizes an object 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": 1,
      },
    },
  },
  "result": Object {
    "user": 1,
  },
}
`;

exports[`ObjectSchema normalization normalizes plain objects as shorthand for ObjectSchema 1`] = `
Object {
  "entities": Object {
    "user": Object {
      "1": Object {
        "id": 1,
      },
    },
  },
  "result": Object {
    "user": 1,
  },
}
`;
